### PR TITLE
Use docker-compose instead of manually stitching docker run together

### PR DESCRIPTION
**Pull request by phiSgr**

### CODE REVIEW

File: apps/framework-cli/src/infrastructure/olap/clickhouse.rs [hunk 1]
```diff
@@ -262,7 +265,7 @@ pub async fn check_ready(
     crate::utilities::retry::retry(
         || run_query(&dummy_query, configured_client),
         |i, e| {
-            i < 10
+            i < 20
```
Comment (phiSgr) [inline apps/framework-cli/src/infrastructure/olap/clickhouse.rs:268]: give it even more time for clickhouse to be ready

File: apps/framework-cli/src/utilities/docker.rs [hunk 2]
```diff
@@ -143,14 +194,54 @@ pub fn remove_network(network_name: &str) -> std::io::Result<String> {
...
+        )
+        .env(
+            "CLICKHOUSE_HOST_PORT",
+            project.clickhouse_config.host_port.to_string(),
+        )
+        .env(
+            "CLICKHOUSE_POSTGRES_PORT",
+            project.clickhouse_config.postgres_port.to_string(),
+        )
+        .env("CLICKHOUSE_VERSION", "24.1.3") // https://github.com/ClickHouse/ClickHouse/issues/60020
```
Comment (phiSgr) [inline apps/framework-cli/src/utilities/docker.rs:177]: downgrading to fix the db not found error.

File: apps/framework-cli/src/cli/routines/validate.rs [hunk 3]
```diff
@@ -104,24 +104,28 @@ impl Routine for ValidatePandaHouseNetwork {
...
-                    "to validate red panda cluster".to_string(),
+                    format!("to validate red panda cluster, {}", err),
                 ),
                 err,
             )
         })?;
 
+        println!("output: {:?}", output);
+
         if output.contains(REDPANDA_CONTAINER_NAME) {
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/routines/validate.rs]: Occasionally it says `localhost`.

```
CLUSTER
=======
redpanda.75be83ff-cd2a-47e0-809d-5544d7d88cb9

BROKERS
=======
ID    HOST       PORT
0*    localhost  19092
```

File: apps/framework-cli/src/cli/routines.rs [hunk 4]
```diff
@@ -116,6 +116,7 @@ pub struct RoutineSuccess {
 
 // Implement success and info contructors and a new constructor that lets the user choose which type of message to display
 impl RoutineSuccess {
+    #[allow(dead_code)]
```
Comment (callicles) [inline apps/framework-cli/src/cli/routines.rs]: Why did we allow dead code here - can you add some color with a comment?

**Review by callicles — APPROVED** [review]
Lgtm

**Pull request merged**